File: FarNet/Plugins.NET/Tutorial-ToolPlugin-VB/Hello.vb
Imports System
Imports FarNet

Public Class VBTool
Inherits ToolPlugin
	Public Overrides Sub Invoke (ByVal sender As Object, ByVal e As ToolEventArgs)
		Far.Msg("Hello " + Name + " " + e.From.ToString())
	End Sub
End Class

File: FarNet/Plugins.NET/Tutorial-ToolPlugin/Hello.cs
// Two lessons:
// 1) How to create a plugin called from Far plugin menus.
// 2) How to build a localized plugin and use localized strings.

using System;
using FarNet;

public class Hello : ToolPlugin
{
	public override void Invoke(object sender, ToolEventArgs e)
	{
		Far.Msg(string.Format(GetString("Format"), GetString("Hello"), GetString("World")));
	}
}
